src/wolff_client.erl

```diff
@@ -314,8 +340,8 @@ is_connected(MaybePid, {Host, Port}) ->
 
 is_alive(Pid) -> is_pid(Pid) andalso erlang:is_process_alive(Pid).
 
-add_conn({ok, Pid}, ConnId, Conns) -> Conns#{ConnId => Pid};
-add_conn({error, Reason}, ConnId, Conns) -> Conns#{ConnId => Reason}.
+add_conn({ok, Pid}, ConnId, Conns) when is_pid(Pid) -> Conns#{ConnId => Pid};
+add_conn({error, Reason}, ConnId, Conns) when not is_pid(Reason) -> Conns#{ConnId => Reason}.
```

Comment:
Is guard expression really needed here? Seems a bit strange for the reason to be a Pid, and this case is not handled either.

Reply:
it's intended to crash when Reason is actually a pid